### DISCUSSION TITLE
Upgrade best-practises, changes to originals directory: Full re-scan necessary?

### DISCUSSION THREAD

**Discussion by mrcalvin** [photoprism/photoprism#2954] [Q&A]
Hi everyone!

First, thanks for this great application. We are very happy to have `photoprism` organize our collections! We run it on our TrueNAS/ FreeBSD system.

I searched for answers in the docs and the previous discussions recorded here, but not no avail, so here my questions:

- If upgrading `photoprism` (using FreeBSD's `port`), does one have to run a full re-scan? Or, is a lightweight rescan sufficient?
- If re-organizing the originals directory (sub-directory structure), does one have to run a full re-scan, or will the changed originals locations be picked up by a lightweight scan also? Say, image files retain their names and are unchanged, but are moved to other directories.

Thx for a clarification!

**Comment by lastzero** [comment]
> * If upgrading `photoprism` (using FreeBSD's `port`), does one have to run a full re-scan? Or, is a lightweight rescan sufficient?

It depends, see the notes for each release:

- https://docs.photoprism.app/release-notes/

If you are not affected by any of the changes, then you don't have to update.

See also the [notes in our getting started](https://docs.photoprism.app/getting-started/docker-compose/#examples) docs that you probably skipped as a FreeBSD user (please help update the related docs you would read as a FreeBSD user):

> `docker compose exec photoprism photoprism index -f` rescans all originals, including already indexed and unchanged files. This may be necessary after major upgrades and after migrations of the database schema, especially if search results are missing or incorrect. Note you can also start a [rescan from the user interface](https://docs.photoprism.app/user-guide/library/originals/) by navigating to Library > Index, checking "Full Rescan" and then clicking "Start".

> * If re-organizing the originals directory (sub-directory structure), does one have to run a full re-scan, or will the changed originals locations be picked up by a lightweight scan also? Say, image files retain their names and are unchanged, but are moved to other directories.

It should be detected when you run a regular scan. Only if you encounter problems that you cannot solve otherwise (i.e. before reporting a bug), please try a rescan and see if it solves the problem.

**Reply by lastzero** [reply]
We have updated our docs for you:

- https://docs.photoprism.app/getting-started/ports/freebsd/#when-should-i-perform-a-complete-rescan
- https://docs.photoprism.app/getting-started/faq/#when-should-i-perform-a-complete-rescan
- https://docs.photoprism.app/getting-started/updates/#complete-rescan

**Reply by mrcalvin** [reply]
@lastzero thx for the swift reply! much appreciated!

A follow-up: Will a full re-scan, in case, retain the face labels we had provided so far (so, match then re-scanned faces given the corpus of previously labelled faces); or, does a full re-scan require us to re-do the manual labeling work. 

Thx!

**Reply by graciousgrey** [reply]
A full re-index does not overwrite manual changes.

**Reply by lastzero** [reply]
We have also added this information to our documentation hoping that it is detailed enough now:
- https://github.com/photoprism/photoprism-docs/commit/f1da764b1d31f7d262508073653c9ef38f1814f4

**Reply by lastzero** [reply]
> We recommend performing a complete rescan after major updates to take advantage of new search filters and sorting options. Be sure to [read the notes for each release](https://docs.photoprism.app/release-notes/) to find out what changes have been made and if they might affect your library, for example, because of the file types you have or because new search features have been added. If you encounter problems that you cannot solve otherwise (i.e. before reporting a bug), please also try a rescan and see if it solves the problem.
>
> You can start a [rescan from the user interface](https://docs.photoprism.app/user-guide/library/originals/) by navigating to Library > Index, selecting "Complete Rescan", and then clicking "Start". Manually entered information such as labels, people, titles or descriptions will not be modified when indexing, even if you perform a "complete rescan".

**Reply by lastzero** [reply]
Updated as well:
- https://docs.photoprism.app/user-guide/library/originals/#when-should-complete-rescan-be-selected
